tests/util/dm_tests.rs

```diff
@@ -40,12 +40,18 @@ pub fn test_linear_device(paths: &[&Path]) -> () {
                                  MIN_MDA_SECTORS,
                                  false)
             .unwrap();
-    let total_blockdev_size: Sectors = initialized.iter().map(|i| i.avail_range().length).sum();
+    let total_blockdev_size: Sectors = initialized
+        .iter()
+        .map(|i| i.borrow().avail_range().1)
+        .sum();
 
     let segments = initialized
         .iter()
-        .map(|block_dev| block_dev.avail_range())
-        .collect::<Vec<Segment>>();
+        .map(|block_dev| {
+                 let (start, length) = block_dev.borrow().avail_range();
```

Comment:
This actions happens repeatedly. Can it be combined into a single utility function?